File: CodeForces/1926/C.cpp
#include <iostream>
#include <vector>

int sum_of_digits(int n) {
    int sum = 0;
    while (n > 0) {
        sum += n % 10;
        n /= 10;
    }
    return sum;
}

int calculate_total_sum(int n, std::vector<int>& memo) {
    int total_sum = 0;
    for (int i = 1; i <= n; i++) {
        total_sum += memo[i];
    }
    return total_sum;
}

int main() {
    std::vector<int> memo(1000001, 0);

    int t;
    scanf("%d", &t);

    for (int i = 0; i < t; i++) {
        int n;
        scanf("%d", &n);
        printf("%d\n", calculate_total_sum(n, memo));
    }

    return 0;
}

File: CodeForces/1926/C.py
def sum_of_digits(n, memo):
    if memo[n] != 0:
        return memo[n]
    else:
        memo[n] = sum(int(digit) for digit in str(n))
        return memo[n]

def calculate_total_sum(n, memo):
    total_sum = 0
    for i in range(1, n+1):
        total_sum += sum_of_digits(i, memo)
    return total_sum

memo = [0] * 1000001

t = int(input())

for _ in range(t):
    n = int(input())
    print(calculate_total_sum(n, memo))
